cimloader/databases/mysql.py

```diff
@@ -0,0 +1,186 @@
+import re
+import mysql.connector
+import logging
+import importlib
+import json
+import enum
+import time
+
+from cimloader.databases import ConnectionInterface, ConnectionParameters, Parameter, QueryResponse
+from cimgraph.data_profile.known_problem_classes import ClassesWithoutMRID
+from cimgraph.models import GraphModel
+
+_log = logging.getLogger(__name__)
+
+class MySQLConnection(ConnectionInterface):
+    def __init__(self, connection_parameters: ConnectionParameters):
+        self.connection_parameters = connection_parameters
+        self.cim_profile = connection_parameters.cim_profile
+        self.cim = importlib.import_module('cimgraph.data_profile.' + self.cim_profile)
+        self.namespace = connection_parameters.namespace
+        self.host = connection_parameters.host
+        self.port = connection_parameters.port
+        self.username = connection_parameters.username
+        self.password = connection_parameters.password
+        self.database = connection_parameters.database
+        self.connection = None
+        self.cursor = None
+
+    def connect(self):
+        if not self.cursor:
+            if not self.database: # Set database name to CIM Profile name if not specified
+                self.database = self.cim_profile 
+            try:
+                self.connection = mysql.connector.connect(host = self.host, user = self.username, password = self.password, database = self.database)
+                self.cursor = self.connection.cursor(buffered = True)
+            except:
+                _log.error('Could not connect to database')
+
+    def disconnect(self):
+        self.cursor = None
```

Comment:
is that all that is needed to properly disconnect from the database? is there something that needs to be done with self.connection here?